File: utils/logger.py
import logging
import os
from datetime import date

def setup_logger(name="lms_logger"):
    log_date = date.today().isoformat()
    log_dir = "logs"
    os.makedirs(log_dir, exist_ok=True)
    log_path = os.path.join(log_dir, f"lms_log_{log_date}.txt")

    logger = logging.getLogger(name)
    logger.setLevel(logging.INFO)

    if not logger.handlers:
        file_handler = logging.FileHandler(log_path)
        formatter = logging.Formatter("%(asctime)s - [Python] %(levelname)s - %(message)s")
        file_handler.setFormatter(formatter)
        logger.addHandler(file_handler)

    return logger

File: electron-app/scripts/utils/logger.js
const fs = require("fs");
const path = require("path");

const date = new Date().toISOString().slice(0, 10);
const logDir = path.join(__dirname, "../logs");
const logPath = path.join(logDir, `lms_log_${date}.txt`);

if (!fs.existsSync(logDir)) fs.mkdirSync(logDir);

function log(message) {
  const timestamp = new Date().toISOString();
  const line = `[${timestamp}] ${message}\n`;

  fs.appendFileSync(logPath, line);
  console.log(message);
}

module.exports = log;

